src/Controller/Component/AuthComponent.php

```diff
@@ -354,7 +354,7 @@ protected function _unauthenticated(Controller $controller): ?Response
         }
         $result = $auth->unauthenticated($controller->getRequest(), $response);
         if ($result !== null) {
-            return $result;
+            return $result instanceof Response ? $result : null;
```

Comment:
Doesn't this technically treat true and false the same?  Should incorrect results be allowed?

Reply:
Yes, but I am just interested in getting things working again rather than make deprecated code perfect :slightly_smiling_face:.

Reply:
No worries, just poking at things that are of interest to me.